src/lib/threadpool.c

```diff
@@ -553,6 +556,15 @@ int pool_init(pool_t *pool,
 	}
 
 	pool_threads_init(pool);
+
+	rc = uv_key_create(&thread_identifier_key);
+	if (rc != 0) {
+		uv_mutex_destroy(&pi->outq_mutex);
+		free(pi);
+		return rc;
+	}
+	uv_key_set(&thread_identifier_key, (void*)main_thread_magic);
```

Comment:
Maybe a silly question but should the key be set to some other value for the child threads?

Reply:
Never mind, I see that `pthread_key_create` guarantees the associated value is initialized to `NULL`.